recipes/flex/all/conanfile.py

```diff
@@ -34,6 +34,11 @@ def source(self):
     def export_sources(self):
         export_conandata_patches(self)
 
+    def requirements(self):
+        # Flex requires M4 to be compiled. If consumer does not have M4
+        # installed, Conan will need to know that Flex requires it.
+        self.requires("m4/1.4.19")
```

Comment:
@RubenRBS @uilianries 
@jcar87 realized when M4 is not found in your system, Conan needs to know that M4 is required to compile flex so it needs to appear as regular requirement as well.

Reply:
@franramirez688  To be built or to be executed? Because M4 is already listed as tool_requires 🤔

Reply:
To be built. As tool_requires, Conan is putting the binary correctly in my PATH, so that's what I need, but if I'm using Docker for instance, and I don't have M4 installed in that image, Conan is not retrieving M4 because it's not even marked as a requirement of Flex, so I need to put it as `requires` as well to prevent that. @jcar87 correct me if I'm wrong.

Reply:
Both :D
* Flex requires `m4` to be built -> so it's a `tool_requires`
* Flex (executable inside the package) requires m4 to run -> so it needs to be a `requires`. This also transitively makes m4 a `tool_requires` for anyone using flex as a tool_requires. 

Confusing, I know :D 

I think the comment could say something along the lines of "the flex executable requires m4 to run"

Reply:
> Flex (executable inside the package) requires m4 to run -> so it needs to be a requires

I don't know what to say ... 😱